sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthRSA256Service.java

```diff
@@ -68,6 +80,82 @@ public AuthRSA256Service(RsaPublicKeyLoader rsaPublicKeyLoader, long leeway) {
+   * same base as the configured {@link KeySource} source uri, that depends on the implementation.
+   *
+   * @param decodedJWT the decoded json web token.
+   * @param loadedPublicKey the structure with all correlated attributes regarding the key.
+   * @return <code>true</code> if the jwt issuer is the same as the configured requiredIssuer and
+   *     the baseUrl is the same as the {@link KeySource} uri or the configuration attribute
+   *     requiredIssuer is not set. Otherwise <code>false</code> is returned.
+   */
+  private boolean validateIssuer(DecodedJWT decodedJWT, LoadedPublicKey loadedPublicKey) {
+
+    if (StringUtils.isNotBlank(loadedPublicKey.getRequiredIssuer())
+        && (!StringUtils.equals(decodedJWT.getIssuer(), loadedPublicKey.getRequiredIssuer())
+            || !validateIssuerAgainstKeySourceUrl(
+                decodedJWT.getIssuer(), loadedPublicKey.getKeySource()))) {
+      LOG.error(
+          "The required issuer {} is not matching the jwt issuer {}",
+          loadedPublicKey.getRequiredIssuer(),
+          decodedJWT.getIssuer());
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Validate, that the provided issuer has the same base url as the keySource uri if set. This is
+   * only for the implementations {@link JwksKeySource} and {@link OpenIdProviderDiscoveryKeySource}
+   *
+   * @param issuer The issuer url.
+   * @param keySource tThe {@link KeySource} that is used to verify it's key uri.
+   * @return <code>true</code> if the {@link KeySource} implemenation key uri is null or the base of
+   *     the {@link KeySource} key uri is the same as the issuer. Otherwise <code>false</code> is
+   *     returned.
+   */
+  private boolean validateIssuerAgainstKeySourceUrl(String issuer, KeySource keySource) {
+    String keySourceUrl;
+
+    if (Objects.requireNonNull(keySource) instanceof JwksKeySource) {
+      JwksKeySource jwksKeySource = (JwksKeySource) keySource;
+      keySourceUrl = jwksKeySource.getJwksUri();
+    } else if (keySource instanceof OpenIdProviderDiscoveryKeySource) {
+      OpenIdProviderDiscoveryKeySource oidcKeySource = (OpenIdProviderDiscoveryKeySource) keySource;
+      keySourceUrl = oidcKeySource.getIssuerUrl();
+    } else {
+      return true;
+    }
+
+    return bothBlankOrHaveSameDomain(keySourceUrl, issuer);
+  }
+
+  /**
+   * validate, if both parameters are empty or have the same domain.
+   *
+   * @param lhsUrl optional nullable {@link String}, that contains a url.
+   * @param rhsUrl optional nullable {@link String}, that contains a url.
+   * @return <code>true</code>, if both arguments are blank or do contain the same domain. Otherwise
+   *     <code>false</code> is returned.
+   */
+  private boolean bothBlankOrHaveSameDomain(String lhsUrl, String rhsUrl) {
+    if (!StringUtils.isAllBlank(lhsUrl, rhsUrl)) {
+      if (StringUtils.isBlank(lhsUrl) ^ StringUtils.isBlank(rhsUrl)) {
+        return false;
+      }
```

Comment:
Would be nice to see the same code like in the documentation:
```
if (StringUtils.isAllBlank(lhsUrl, rhsUrl)) {
  return true;
}
...
```

Reply:
I'm not sure about you intention. Is the early exit what you ment?

Reply:
Yes. I like early exits, because it makes your code more readable IMHO. It also helps with the "keep left" rule.

Reply:
finally I removed this line because it was covered by the business logic before.